api/src/services/settings.js

```diff
@@ -57,6 +62,8 @@ module.exports = {
         }
         if (replace) {
           doReplace(doc.settings, body);
+        } else if (overwrite) {
+          doc.settings = body;
```

Comment:
Maybe put the overwrite check first? If someone specifies `overwrite` and `replace` I'd expect the overwrite to win.